jfix-zookeeper/src/test/kotlin/ru/fix/zookeeper/lock/PersistentExpiringLockManagerTest.kt

```diff
@@ -311,4 +343,59 @@ internal class PersistentExpiringLockManagerTest {
     private fun nodeExists(path: String) = zkServer.client.checkExists().forPath(path) != null
 
     private fun zkTree() = ZkTreePrinter(zkServer.client).print("/")
-}
\ No newline at end of file
+
+    /**
+     * @param latchReleaseOperation - latch that lock main thread before manager.release operation
+     * @param latchMainThread - latch that lock main thread for waiting scheduler process locks (one lock this case)
+     * @param prolongExceptions - set for adding any exceptions when prolong lock in scheduler
+     * @param lockProlongStatus - result of prolonging lock in schdeduler
+     *
+     * @return manager with scheduler that release latchReleaseOperation latch, wait a little
+     * and release latchMainThread latch. This concurrency sequence work once and for one lock in manager
+     */
+    private fun createScheduleLatchedManager(
+            latchReleaseOperation: CountDownLatch,
+            latchMainThread: CountDownLatch,
+            prolongExceptions: ConcurrentHashMap.KeySetView<Exception, Boolean>,
+            lockProlongStatus: AtomicBoolean
+    ): PersistentExpiringLockManager {
+
+        val managerConfig = DynamicProperty.of(PersistentExpiringLockManagerConfig(
+                lockCheckAndProlongInterval = Duration.ofSeconds(1)
+        ))
+        val locksContainer = ActiveLocksContainer()
+        val scheduler = NamedExecutors.newSingleThreadScheduler(
+                "test-PersistentExpiringLockManager", NoopProfiler()
+        )
+        scheduler.schedule(
+                Schedule.withDelay(managerConfig.map { it.lockCheckAndProlongInterval.toMillis() }),
+                0
+        ) {
+            locksContainer.processAllLocks { _, lock, _ ->
+                var prolonged = false
+                try {
+                    latchReleaseOperation.countDown()
+                    TimeUnit.MILLISECONDS.sleep(50) // wait a little for trying release lock in parallel
+                    prolonged = lock.checkAndProlongIfExpiresIn(
```

Comment:
You are copy-pasting big part of production code to tests, increasing coupling. 
You braking incapacitation of Lock manager by extracting scheduler of manager outside through public api. 
Which behavior you are testing, managers or LocksContainers? 
If Managers, what case or possible problem you care to reproduce, that it is possible to release lock in the middle of the prolongation process?  You have two choices: 
A) Setup small locks prolongation timeout, create 100 locks, launch separate process that releases half of 50 locks concurrently with some delay. Check that locks released successfully and there are no errors. 
B)  Extend API of Manager, e.g. 
```
Manager.addLockProlongationAspect(LockProlongationAspect)
LockProlongationAspect{
   beforeLockProlongation(lockId)
   afterLockProlongation(lockId)
}
```
and use that kind of aspect in your Tests. 

Right know Test is testing Test code, and duplicating manager logic in test is inadequate to the testing purpose